src/server/renderer.js

```diff
@@ -44,14 +44,14 @@ export default (projectConfig, projectToolsConfig) => {
           console.error('ROUTER ERROR:', pretty.render(error));
           res.status(500);
         } else if (renderProps) {
-          loadPropsOnServer(renderProps, (err, asyncProps) => {
+          loadOnServer(renderProps, store).then(() => {
```

Comment:
@sompylasar , to be honest , there I cannot see the situations when this promise could be failed.
See: we need to wait until all our async connections are resolver or failed.. And it is..
So if some promise somewhere is failed - the result will be stored to redux state and you can react on such situations.
Here we just need to wait until everything is resolved or rejected.
The same in Async Props, BTW... https://github.com/rackt/async-props/issues/41
May be you can suggest something?
May be we need to change interface here a little bit and use callback - even without error argument?